### ISSUE TITLE
Difference between n_samples and n_iter?

### ISSUE THREAD

**Issue by Neosettler** [CompVis/stable-diffusion#218]
greetings, would anyone care to explain what is the difference between n_samples and n_iter? they both make n output images, it's a bit confusing.

Thank you,

**Comment by RiccardoRiglietti** [comment]
```
  --n_samples N_SAMPLES
                        how many samples to produce for each given prompt. A.k.a batch size

```
This means how many images will be generated.


`  --n_iter N_ITER       sample this often
`

number of denoising iterations for each image.

**Comment by Neosettler** [comment]
thank you Riccardo, n_iter generates an image for each iteration, is that normal?

**Comment by BeaverInGreenland** [comment]
n_iter generates n_samples of images for each iteration

**Comment by Neosettler** [comment]
Thank yo for your input Beaver, 

What happens in the background for n_iter? does it re-run the exact same parameters, same seed, in hope for a variation?
What is the difference for each n_iter?

The same question could be asked for n_samples, if this is not too much to ask.

Note: maybe someone could point out where to look in the code base?

Thank you,

**Comment by KostyaAtarik** [comment]
`--n_samples A.k.a batch size` is literally a batch size, increasing it you will try to fit more and more data to your GPU for a single run, increasing `n_iter` on the other hand just increases number of batches (each of size `--n_samples`) being processed sequentially.
Seed is different for each generated image by default, you can use fixed seed by passing `--fixed_code` parameter to the `txt2img.py` script (seed will be fixed for different iterations the number of which is controlled by the `n_iter` parameter, images in the batch will have different seeds, so if you want to generate `n` images using a single seed set `--n_samples` to `1` and `n_iter` to `n`).

**Comment by Neosettler** [comment]
Thank you for your explanation Kostya. 

Using `--fixed_code`, if the seed is the same for each images, what other parameters are being randomized, if any?

Dang, I'm going to laydown for a minute if you don't mind, I just got some serious headache all of a sudden. :)

**Comment by ironharvy** [comment]
> --n_samples A.k.a batch size is literally a batch size, increasing it you will try to fit more and more data to your GPU for a single run, increasing n_iter on the other hand just increases number of batches (each of size --n_samples) being processed sequentially.

Does this mean that higher --n_samples can potentially produce better result at the expense of GPU memory?

**Comment by KostyaAtarik** [comment]
No, it can't. I think here you confuse training phase (when changing batch size actually can drastically affect resulting model quality) with inference phase (when changing batch size only affects speed and memory consumption, whereas the generation result is determined by the fixed weights of the trained model + some randomness in case of stochastic sampler usage).

**Comment by LemonCoding** [comment]
> `--n_samples A.k.a batch size` is literally a batch size, increasing it you will try to fit more and more data to your GPU for a single run, increasing `n_iter` on the other hand just increases number of batches (each of size `--n_samples`) being processed sequentially. Seed is different for each generated image by default, you can use fixed seed by passing `--fixed_code` parameter to the `txt2img.py` script (seed will be fixed for different iterations the number of which is controlled by the `n_iter` parameter, images in the batch will have different seeds, so if you want to generate `n` images using a single seed set `--n_samples` to `1` and `n_iter` to `n`).

How can we know the seed for each images in a batch. I notice that is not a simple +1, +2
